src/SonarScanner.MSBuild.PreProcessor/AutomaticBaseBranchDetection.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace SonarScanner.MSBuild.PreProcessor
+{
+    internal class AutomaticBaseBranchDetection
+    {
+        private static readonly List<Tuple<string, string>> Candidates = new()
```

Comment:
Not a big fan of this syntax, but I can't use deconstruction or syntactic sugar from later versions of C#, because of net46.
:(

Reply:
Refactored it to a POCO class